pkg/config/stepmeta.go

```diff
@@ -427,6 +432,24 @@ func OptionsAsStringSlice(options []Option) []string {
 	return e
 }
 
+func putStringIfNotEmpty(config map[string]interface{}, key, value string) {
+	if value != "" {
+		config[key] = value
+	}
+}
+
+func putMapIfNotEmpty(config map[string]interface{}, key string, value map[string]string) {
```

Comment:
[curiosity] does the value map always contain strings as values?

Reply:
yes, [that is how p is defined on line 266: `p := map[string]interface{}{}`](https://github.com/SAP/jenkins-library/pull/2406/files/4d4f868b9ec0bec441bb9b05cd7bddf4926ac288#diff-b087135406a2ea97103f35c2827188829d5e9fa4d30a718108364b6b94b6a639R266)